peps/pep-0793.rst

```diff
@@ -0,0 +1,606 @@
+It should be moved to a HOWTO in the documentation.
+
+#. Scan your code for uses of ``PyModule_GetDef``. This function will
+   return ``NULL`` for modules that use the new mechanism. Instead:
+
+   * For getting the contents of the module's ``PyModuleDef``, use the C struct
+     directly. Alternatively, get attributes from the module using, for
+     example, ``PyModule_GetNameObject``, the ``__doc__`` attribute, and
+     ``PyModule_GetStateSize``.
+     (Note that Python code can mutate a module's attributes.)
+   * For testing if a module object is “yours”, use ``PyModule_GetToken``
+     instead.
+     Later in this guide, you'll set the token to *be* the existing
+     ``PyModuleDef`` structure.
+
+#. Scan your code for uses of ``PyType_GetModuleByDef``, and replace them by
+   ``PyType_GetModuleByToken``.
+   Later in this guide, you'll set the token to *be* the existing
+   ``PyModuleDef`` structure.
+
+#. Look at the function identified by ``Py_mod_create``, if any.
+   Make sure that it does not use its second argument (``PyModuleDef``),
+   as it will be called with ``NULL``.
+   Instead of the argument, use the existing ``PyModuleDef`` struct directly.
+#. If using multiple ``Py_mod_create`` slots, consolidate them: pick one of
+   the functions, or write a new one, and call the others from it.
+   Remove all but one ``Py_mod_create`` slots.
+#. Make a copy of the existing ``PyModuleDef_Slot`` array pointed to by
+   the ``m_slots`` member of your ``PyModuleDef``. If you don't have an
+   existing slots array, create one like this:
+
+   .. code-block:: c
+
+    static PyModuleDef_Slot module_slots[] = {
+        {0}
+    };
+
+   Give this array a unique name.
+   Further examples will assume that you've named it ``module_slots``.
+
+#. Add slots for all members of the existing ``PyModuleDef`` structure.
+   See :ref:`pep793-api-summary` for a list of the new slots.
+   For example, to add a name and docstring:
+
+   .. code-block:: c
+
+    static PyModuleDef_Slot module_slots[] = {
+        {Py_mod_name, "mymodule"},
+        {Py_mod_doc, (char*)PyDoc_STR("my docstring")},
+        // ... (keep existing slots here)
+        {0}
+    };
+
+#. If you switched from ``PyModule_GetDef`` to ``PyModule_GetToken``,
+   and/or from ``PyType_GetModuleByDef`` to ``PyType_GetModuleByToken``,
+   add a ``Py_mod_token`` slot pointing to the existing ``PyModuleDef`` struct:
+
+   .. code-block:: c
+
+    static PyModuleDef_Slot module_slots[] = {
+        // ... (keep existing slots here)
+        {Py_mod_token, your_module_def},
+        {0}
+    };
+
+
+#. Add a new export hook.
+
+   .. code-block:: c
+
+    PyMODEXPORT_FUNC PyModExport_examplemodule(PyObject);
+
+    PyMODEXPORT_FUNC
+    PyModExport_examplemodule(PyObject *spec)
+    {
+        return module_slots;
+    }
+
+The new export hook will be used on Python 3.15 and above.
+Once your module no longer supports lower versions, delete the ``PyInit_``
+function and any unused data.
```

Comment:
Perhaps mention that `Py_mod_token` can be deleted at this point?

Reply:
It was only added if it's used, so it would need a bigger clean-up -- and I think it's situation-specific (and rare) enough at this point that you should be looking at the code, not the guide.